draft-ietf-mls-protocol.md

```diff
@@ -2233,6 +2248,15 @@ struct {
         case commit:
           Commit commit;
     }
+
+    select (MLSPlaintextTBS.sender.sender_type) {
+        case member:
+            GroupContext context;
+
+        case preconfigured:
+        case new_member:
+            struct{};
+    }
```

Comment:
I don't love the aesthetics here.  Maybe we could solve this a little more elegantly by having a `context` input to `SignWithLabel`/`VerifyWithLabel`?

Alternatively: Do we even need this field any more?  This field was added before `membership_tag`, and now the `membership_tag` already (a) demonstrates knowledge of the GroupContext and (b) prevents a non-member from synthesizing a valid MLSPlaintext.

Reply:
I not sure the alternative is a good idea for several reasons, for example the `membership_tag` is present only in `MLSPlaintext` and not in `MLSCiphertext`, also I'm not confident in removing things from the signature without a solid proof that it is not necessary (for example, signing the tree hash seems like a reasonable thing to do?).

I feel like adding a `context` input is a bit weird. What do you think about this third option: adding an `option<GroupContext> group_context;` and say something like "group_context MUST contain a value if sender_type = ..." below? This pattern already exists for `confirmation_tag` for example.